api/rest/client/methods.go

```diff
@@ -172,3 +178,72 @@ func (c *Client) GetConnectGraph() (api.ConnectGraphSerial, error) {
 	err := c.do("GET", "/health/graph", nil, &graphS)
 	return graphS, err
 }
+
+// WaitFor is a utility function that allows for a caller to
+// wait for a paticular status for a CID. It returns a channel
+// upon which the caller can wait for the targetStatus.
+func (c *Client) WaitFor(ctx context.Context, ci *cid.Cid, local bool, target api.TrackerStatus, checkFreq time.Duration) (<-chan api.TrackerStatus, <-chan error) {
+	statusCh := make(chan api.TrackerStatus)
+	errCh := make(chan error)
+
```

Comment:
Make sure channels are closed when the goroutine stops (because of error, because of completion)